samples/cpp/face_match.cpp

```diff
@@ -0,0 +1,101 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "opencv2/dnn.hpp"
+#include "opencv2/imgproc.hpp"
+#include "opencv2/highgui.hpp"
+
+#include <iostream>
+
+#include "opencv2/objdetect.hpp"
+
+
+using namespace cv;
+using namespace std;
+
+
+int main(int argc, char ** argv)
+{
+    if (argc != 5)
+    {
+        std::cerr << "Usage " << argv[0] << ": "
+                  << "<det_onnx_path> "
+                  << "<reg_onnx_path> "
+                  << "<image1>"
+                  << "<image2>\n";
+        return -1;
+    }
+
+    String det_onnx_path = argv[1];
+    String reg_onnx_path = argv[2];
+    String image1_path = argv[3];
+    String image2_path = argv[4];
+    std::cout<<image1_path<<" "<<image2_path<<std::endl;
+    Mat image1 = imread(image1_path);
+    Mat image2 = imread(image2_path);
+
+    float score_thresh = 0.9;
+    float nms_thresh = 0.3;
+    float cosine_similar_thresh = 0.34;
+    float l2norm_similar_thresh = 1.32;
+    int top_k = 5000;
+
+    // Initialize DNNFaceDetector
+    Ptr<DNNFaceDetector> faceDetector;
+    
+    faceDetector = DNNFaceDetector::create(det_onnx_path, image1.size(), score_thresh, nms_thresh, top_k);
+    Mat faces_1;
+    faceDetector->detect(image1, faces_1);
+    if (faces_1.rows < 1)
+    {
+        std::cerr << "Cannot find a face in " << image1_path << "\n";
+        return -1;
+    }
+
+    faceDetector = DNNFaceDetector::create(det_onnx_path, image2.size(), score_thresh, nms_thresh, top_k);
+    Mat faces_2;
+    faceDetector->detect(image2, faces_2);
+    if (faces_2.rows < 1)
+    {
+        std::cerr << "Cannot find a face in " << image2_path << "\n";
+        return -1;
+    }
+
+    // Initialize DNNFaceRecognizer
+    Ptr<DNNFaceRecognizer> faceRecognizer = DNNFaceRecognizer::create(reg_onnx_path);
+
+
+    Mat aligned_face1, aligned_face2;
+    faceRecognizer->AlignCrop(image1, faces_1.row(0), aligned_face1);
+    faceRecognizer->AlignCrop(image2, faces_2.row(0), aligned_face2);
+
+    Mat feature1, feature2;
+    faceRecognizer->facefeature(aligned_face1, feature1);
+    faceRecognizer->facefeature(aligned_face1, feature2);
```

Comment:
aligned_face2 ;)

Reply:
Sorry for the late response. I was just notified that something was wrong with the dev machine used by the other contributor @crywang . He is also in this thread and responsible for the part of face recognition. I think he will be soon online for fixing bugs in one or two days.